packages/ui/src/common/components/forms/FilterBox/FilterSearchBox.tsx

```diff
@@ -40,8 +41,10 @@ interface SearchBoxProps extends ControlProps<string> {
   displayReset?: boolean
 }
 export const SearchBox = React.memo(({ value, onApply, onChange, label, displayReset }: SearchBoxProps) => {
+  const debouncedValue = useDebounce(value, 400)
   const change = onChange && (({ target }: ChangeEvent<HTMLInputElement>) => onChange(target.value))
   const keyDown = onApply && (({ key }: React.KeyboardEvent) => key === 'Enter' && onApply())
+  const isValid = () => !debouncedValue || debouncedValue.length === 0 || debouncedValue.length > 2
```

Comment:
Similarly here the sidebar should only open when value is valid:
```suggestion
  const isValid = () => !debouncedValue || debouncedValue.length === 0 || debouncedValue.length > 2
  const keyDown = !isValid || !onApply ? undefined : (({ key }: React.KeyboardEvent) => key === 'Enter' && onApply())
```